src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1523,6 +1665,441 @@ const Vector3f Ekf2::get_vel_body_wind()
+			}
+		}
+
+		// normalise the weights
+		if (sum_of_vpos_weights > 0.0f) {
+			for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+				vpos_blend_weights[i] = vpos_blend_weights[i] / sum_of_vpos_weights;
+			}
+
+			sum_of_all_weights += 1.0f;
+		};
+	}
+
+	// calculate an overall weight
+	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+		_blend_weights[i] = (hpos_blend_weights[i] + vpos_blend_weights[i] + spd_blend_weights[i]) / sum_of_all_weights;
+	}
+
+	return true;
+}
+
+/*
+ Update the internal state estimate for a blended GPS solution that is a weighted average of the phsyical receiver solutions
+ with weights are calculated in calc_gps_blend_weights(). This internal state cannot be used directly by estimators
+ because if physical receivers have significant position differences,  variation in receiver estimated accuracy will
+ cause undesirable variation in the position soution. Offsets are handled in the calc_gps_blend_outputs() function.
+*/
+void Ekf2::update_gps_blend_states(void)
+{
+	// initialise the blended states so we can accumulate the results using the weightings for each GPS receiver.
+	_gps_state[GPS_BLENDED_INSTANCE].time_usec = 0;
+	_gps_state[GPS_BLENDED_INSTANCE].lat = 0;
+	_gps_state[GPS_BLENDED_INSTANCE].lon = 0;
+	_gps_state[GPS_BLENDED_INSTANCE].alt = 0;
+	_gps_state[GPS_BLENDED_INSTANCE].fix_type = 0;
+	_gps_state[GPS_BLENDED_INSTANCE].eph = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].epv = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].sacc = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].vel_m_s = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].vel_ned[0] = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].vel_ned[1] = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].vel_ned[2] = 0.0f;
+	_gps_state[GPS_BLENDED_INSTANCE].vel_ned_valid = true;
+	_gps_state[GPS_BLENDED_INSTANCE].nsats = 0;
+	_gps_state[GPS_BLENDED_INSTANCE].gdop = 0.0f;
+
+	_blended_antenna_offset.zero();
+
+	// combine the the GPS states into a blended solution using the weights calculated in calc_blend_weights()
+	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+		// blend the timing data
+		_gps_state[GPS_BLENDED_INSTANCE].time_usec += (uint64_t)((double)_gps_state[i].time_usec * (double)_blend_weights[i]);
+
+		// use the highest status
+		if (_gps_state[i].fix_type > _gps_state[GPS_BLENDED_INSTANCE].fix_type) {
+			_gps_state[GPS_BLENDED_INSTANCE].fix_type = _gps_state[i].fix_type;
+		}
+
+		// calculate a blended average speed and velocity vector
+		_gps_state[GPS_BLENDED_INSTANCE].vel_m_s += _gps_state[i].vel_m_s * _blend_weights[i];
+		_gps_state[GPS_BLENDED_INSTANCE].vel_ned[0] += _gps_state[i].vel_ned[0] * _blend_weights[i];
+		_gps_state[GPS_BLENDED_INSTANCE].vel_ned[1] += _gps_state[i].vel_ned[1] * _blend_weights[i];
+		_gps_state[GPS_BLENDED_INSTANCE].vel_ned[2] += _gps_state[i].vel_ned[2] * _blend_weights[i];
+
+		// report the best valid accuracies and DOP metrics
+
+		if (_gps_state[i].eph > 0.0f
+		    && _gps_state[i].eph < _gps_state[GPS_BLENDED_INSTANCE].eph) {
```

Comment:
We need to take the smallest non zero value so that won't work either. I need to initialise _gps_state[GPS_BLENDED_INSTANCE].eph to the largest float.

Reply:
fixed